deepchem/models/tensorgraph/tensor_graph.py

```diff
@@ -106,7 +106,26 @@ def _add_layer(self, layer):
     for in_layer in layer.in_layers:
       self._add_layer(in_layer)
       self.nxgraph.add_edge(in_layer.name, layer.name)
-
+  
+  def _eval_valid(self, sess, valid_fdg):
```

Comment:
Oh yeah that's much neater, I'll try that now. I just thought of this because it uses the loss directly.